pkg/skaffold/deploy/helm.go

```diff
@@ -308,18 +310,43 @@ func (h *HelmDeployer) deployRelease(ctx context.Context, out io.Writer, r lates
 		return nil, fmt.Errorf("release args: %w", err)
 	}
 
-	iErr := h.exec(ctx, out, r.UseHelmSecrets, args...)
+	err = h.exec(ctx, out, r.UseHelmSecrets, args...)
+	if err != nil {
+		return nil, fmt.Errorf("install: %w", err)
+	}
+
+	b, err := h.getRelease(ctx, r, helmVersion, releaseName, opts.namespace)
+	if err != nil {
+		return nil, fmt.Errorf("get release: %w", err)
+	}
+
+	artifacts := parseReleaseInfo(opts.namespace, bufio.NewReader(&b))
+	return artifacts, nil
+}
 
+// getRelease confirms that a release is visible to helm
+func (h *HelmDeployer) getRelease(ctx context.Context, r latest.HelmRelease, helmVersion semver.Version, releaseName string, namespace string) (bytes.Buffer, error) {
+	// Retry, because under Helm 2, at least, a release may not be immediately visible
+	opts := backoff.NewExponentialBackOff()
+	opts.MaxElapsedTime = 4 * time.Second
 	var b bytes.Buffer
 
-	// Be accepting of failure
-	if err := h.exec(ctx, &b, false, getArgs(helmVersion, releaseName, opts.namespace)...); err != nil {
-		logrus.Warnf(err.Error())
-		return nil, nil
+	err := backoff.Retry(
+		func() error {
+			if err := h.exec(ctx, &b, false, getArgs(helmVersion, releaseName, namespace)...); err != nil {
+				logrus.Debugf("unable to get release: %v (will retry):\n%s", err, b.String())
+				return err
+			}
+			return nil
+		}, opts)
+
+	if err == nil {
+		logrus.Debugf("%q release confirmed", releaseName)
+	} else {
+		logrus.Debugf("%q release never appeared: %v", releaseName, err)
```

Comment:
This seems worthy of WARN?

Reply:
I chose not to warn because in general, you should either log messages or pass them upwards, otherwise you'll end up with duplicated log entries. I chose just to surface it upwards.